### PR TITLE
Better finding of the Dropbox location.

### PR DESCRIPTION
**Pull request by billev**
With this change we now use the Dropbox configuration to find the user's Dropbox installation.
Additionally, one can use the environment variable 'dropbox' to set up a temporary override,
useful for testing.

Dropbox is used as the "global shared directory", and is part of the ACMConfiguration. To simplify
setting up the configuration, it was converted from a collection of static properties to a
singleton shared instance; adding the 'getInstance()' calls is the bulk of the lines changed
(though the least interesting part).

The actual Dropbox finding code is added in utils/DropboxFinder.java, as well as unit tests for that
new code.

With junit and mockito/powermock integrated into the project, it will be easier for future
commits to also include unit tests. They will now be expected :)

### CODE REVIEW

File: acm/ivy.xml [hunk 1]
```diff
@@ -33,7 +33,10 @@
     <dependency org="com.seaglasslookandfeel" name="seaglasslookandfeel" rev="0.2.0-JAVA8"/>
     <dependency org="org.swinglabs" name="swingx" rev="1.6.1"/>
     <dependency org="args4j" name="args4j" rev="2.0.16"/>
-    <dependency org="commons-io" name="commons-io" rev="2.4"></dependency>
-    <dependency org="net.sf.opencsv" name="opencsv" rev="2.3"></dependency>
+    <dependency org="commons-io" name="commons-io" rev="2.4"/>
+    <dependency org="net.sf.opencsv" name="opencsv" rev="2.3"/>
+    <dependency org="org.powermock" name="powermock-api-mockito" rev="1.6.4" />
```
Comment (michibusch) [inline acm/ivy.xml]: Do you think it's possible to force ivy to include these powermock dependencies only during the test target? Keeping the acm deploy jar file small in size is preferable due to slow internet connections in Ghana.

File: acm/src/main/java/org/literacybridge/acm/config/ControlAccess.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ public void init() {
 			@SuppressWarnings("unused")
 			LockACM l = new LockACM(config);
 		} catch (RuntimeException e) {
-			if (!ACMConfiguration.isDisableUI())
+			if (!ACMConfiguration.getInstance().getInstance().isDisableUI())
```
Comment (michibusch) [inline acm/src/main/java/org/literacybridge/acm/config/ControlAccess.java]: Duplicate getInstance()

File: acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        DropboxFinder dbFinder = new DropboxFinder();
+        File infoJson = dbFinder.getInfoFile();
+        return dbFinder.getDropboxPathFromInfoFile(infoJson);
+    }
+
+    /**
+     * Determine if running on Windows.
+     * @return True if running on Windows. False otherwise.
+     */
+    protected boolean onWindows() {
```
Comment (michibusch) [inline acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java]: Use org.literacybridge.acm.utils.OSChecker.java?

File: acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import org.codehaus.jackson.map.ObjectMapper;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+/**
+ * Created by bill on 2/24/16.
+ */
+public class DropboxFinder {
```
Comment (michibusch) [inline acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java]: Out of curiosity: Does the Dropbox Java SDK maybe provide similar functionality?

File: acm/ivy.xml (hunk 1 above)
Comment (billev) [inline acm/ivy.xml]: That's a great idea. I haven't been able to make Ivy/Ant make it work, though. It does seem to be just what Ivy should be good for.

I'll continue to work on that, but I may deploy it initially without. It is "only" a little under 2MB.

File: acm/src/main/java/org/literacybridge/acm/config/ControlAccess.java (hunk 2 above)
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/config/ControlAccess.java]: Thanks.

File: acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java (hunk 3 above)
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java]: Good call.

File: acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java (hunk 4 above)
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/utils/DropboxFinder.java]: I couldn't find that in the API. I can see, from their point of view, it isn't actually something I should care about.

**Pull request closed without merging**